src/k2/transport/TXEndpoint.cpp

```diff
@@ -72,18 +72,18 @@ std::unique_ptr<TXEndpoint> TXEndpoint::fromURL(const String& url, BinaryAllocat
         }
         ip = seastar::rdma::EndPoint::GIDToString(tmpip6);
     }
-    return std::make_unique<TXEndpoint>(std::move(protocol), std::move(ip), port, std::move(allocator));
+    return std::make_unique<TXEndpoint>(std::move(protocol), std::move(ip), port, allocator);
 }
 
 TXEndpoint::~TXEndpoint() {
     K2LOG_D(log::tx, "dtor");
 }
 
-TXEndpoint::TXEndpoint(String&& pprotocol, String&& pip, uint32_t pport, BinaryAllocatorFunctor&& allocator):
+TXEndpoint::TXEndpoint(String&& pprotocol, String&& pip, uint32_t pport, BinaryAllocator allocator):
```

Comment:
receive the allocator by && to avoid a copy

Reply:
done. But we need to manually copy the allocator in some object for a shared allocator to avoid making the original allocator as empty.